web/docs/formats/syslog.md

```diff
@@ -0,0 +1,66 @@
+# syslog
+
+Reads syslog messages.
+
+## Synopsis
+
+```
+syslog
+```
+
+## Description
+
+Syslog is a standard format for message logging.
+Tenzir supports reading syslog messages in both the standardized "Syslog Protocol" format
+([RFC 5424](https://tools.ietf.org/html/rfc5424)), and the older "BSD syslog Protocol" format
+([RFC 3164](https://tools.ietf.org/html/rfc3164)).
+
+Depending on the syslog format, the result can be different.
+Here's an example of a syslog message in RFC 5424 format:
+
+```
+<165>8 2003-10-11T22:14:15.003Z mymachineexamplecom evntslog - ID47 [exampleSDID@32473 iut="4" eventSource="Applic\"ation" eventID="1011"] BOMAn application event log entry...
+```
+
+With this input, the parser will produce the following output, with the schema name `syslog.rfc5424`:
+
+```json
+{
+  "facility": 20,
+  "severity": 5,
+  "version": 8,
+  "ts": "2003-10-11T22:14:15.003000",
+  "hostname": "mymachineexamplecom",
+  "app_name": "evntslog",
+  "process_id": "",
+  "message_id": "ID47",
+  "message": "BOMAn application event log entry..."
+}
+```
+
+Here's an example of a syslog message in RFC 3164 format:
+
+```
+<34> Nov 16 14:55:56 mymachine PROGRAM: Freeform message
+```
+
+With this input, the parser will produce the following output, with the schema name `syslog.rfc3164`:
+
+```json
+{
+  "facility": 4,
+  "severity": 2,
+  "timestamp": "Nov 16 14:55:56",
+  "host": "mymachine",
+  "tag": "PROGRAM",
+  "content": "Freeform message"
```

Comment:
I'm curious: Is there a reason why it's "content" for RFC 3164, and "message" RFC 5424?

Reply:
That's following the RFCs: in 3164, the `message` field is divided into two parts, `tag` and `content`. In 5424, the information previously conveyed by the tag is given in the header. To quote Wikipedia:

> In RFC 3164, the message component (known as MSG) was specified as having these fields: TAG, which should be the name of the program or process that generated the message, and CONTENT which contains the details of the message. 
> 
> Described in RFC 5424, "MSG is what was called CONTENT in RFC 3164. The TAG is now part of the header, but not as a single field. The TAG has been split into APP-NAME, PROCID, and MSGID. This does not totally resemble the usage of TAG, but provides the same functionality for most of the cases."

Reply:
That's following the RFCs: in 3164, the `message` field is divided into two parts, `tag` and `content`. In 5424, the information previously conveyed by the tag is given in the header. To quote Wikipedia:

> In RFC 3164, the message component (known as MSG) was specified as having these fields: TAG, which should be the name of the program or process that generated the message, and CONTENT which contains the details of the message. 
> 
> Described in RFC 5424, "MSG is what was called CONTENT in RFC 3164. The TAG is now part of the header, but not as a single field. The TAG has been split into APP-NAME, PROCID, and MSGID. This does not totally resemble the usage of TAG, but provides the same functionality for most of the cases."

Reply:
Thanks for the explanation, it does make sense to follow the RFCs here. Should we perhaps add an info box to the docs explaining this (I'm not sure)?